### PR TITLE
Fields of Study

### PR DESCRIPTION
**Pull request by cesar82**

### CODE REVIEW

File: src/components/molecules/CIPCodeSelector/index.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from 'react';
+import uuid from 'uuid/v4';
+import { StyledTypeahead, CIPCodeContent } from './style';
+import { useCIPCode } from 'hooks/useCIPCode';
+
+const CIPCodeSelector = ({ schoolType, ...innerProps }, ref) => {
+  const { dataCIPCodes, setParams } = useCIPCode();
+  const [cipCodes, setCIPCodes] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [page, setPage] = useState(1);
```
Comment (normandanielmq) [inline src/components/molecules/CIPCodeSelector/index.js:10]: Pagination is not required here

File: src/components/molecules/CIPCodeSelector/index.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    if (dataCIPCodes) {
+      setCIPCodes(dataCIPCodes);
+      setLoading(false);
+    }
+  }, [dataCIPCodes]);
+
+  const handleSearch = (query) => {
+    setLoading(true);
+    setQuery(query);
+    setPage(1);
```
Comment (normandanielmq) [inline src/components/molecules/CIPCodeSelector/index.js:23]: remove

File: src/components/molecules/CIPCodeSelector/index.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import uuid from 'uuid/v4';
+import { StyledTypeahead, CIPCodeContent } from './style';
+import { useCIPCode } from 'hooks/useCIPCode';
+
+const CIPCodeSelector = ({ schoolType, ...innerProps }, ref) => {
+  const { dataCIPCodes, setParams } = useCIPCode();
+  const [cipCodes, setCIPCodes] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [page, setPage] = useState(1);
+  const [query, setQuery] = useState();
```
Comment (normandanielmq) [inline src/components/molecules/CIPCodeSelector/index.js:11]: We might don't need this after all

File: src/components/molecules/CIPCodeSelector/index.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  useEffect(() => {
+    if (dataCIPCodes) {
+      setCIPCodes(dataCIPCodes);
+      setLoading(false);
+    }
+  }, [dataCIPCodes]);
+
+  const handleSearch = (query) => {
+    setLoading(true);
+    setQuery(query);
```
Comment (normandanielmq) [inline src/components/molecules/CIPCodeSelector/index.js:22]: remove

File: src/components/molecules/CIPCodeSelector/index.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  return (
+    <StyledTypeahead
+      id={uuid()}
+      isLoading={loading}
+      options={cipCodes}
+      labelKey="description"
+      onSearch={handleSearch}
+      paginate
```
Comment (normandanielmq) [inline src/components/molecules/CIPCodeSelector/index.js:34]: remove

File: src/components/pages/ProgramDetailsPage/FieldsOfStudy/index.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        {widgets}
+      </SingleCriteriaList>
+    </Criteria>
+  );
+};
+
+export default FieldsOfStudy;
+
+// fieldsOfStudy: [{fieldName: "Medical Insurance Coding Specialist/Coder.", CIPCode: "51.0713"}]
+// 0: {fieldName: "Medical Insurance Coding Specialist/Coder.", CIPCode: "51.0713"}
```
Comment (normandanielmq) [inline src/components/pages/ProgramDetailsPage/FieldsOfStudy/index.js:98]: Is this necessary?

File: src/components/molecules/CIPCodeSelector/index.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, useState } from 'react';
+import uuid from 'uuid/v4';
+import { StyledTypeahead, CIPCodeContent } from './style';
+import { useCIPCode } from 'hooks/useCIPCode';
+
+const CIPCodeSelector = ({ schoolType, ...innerProps }, ref) => {
+  const { dataCIPCodes, setParams } = useCIPCode();
+  const [cipCodes, setCIPCodes] = useState([]);
+  const [loading, setLoading] = useState(false);
```
Comment (normandanielmq) [inline src/components/molecules/CIPCodeSelector/index.js:9]: Don't needed, use the one comming from the hook

File: src/components/pages/ProgramDetailsPage/FieldsOfStudy/index.js [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+  const widgets = fields.map(({ id }, index) => {
+    const fieldName = `${section}[${index}].fieldName`;
+    const cipCode = `${section}[${index}].CIPCode`;
+    const watchedRow = watch(`${section}[${index}]`);
+
+    return (
+      <div key={id}>
+        <>
+          {' '}
```
Comment (normandanielmq) [inline src/components/pages/ProgramDetailsPage/FieldsOfStudy/index.js:44]: What is this for?

**Review by normandanielmq — APPROVED** [review]

**Review by fabian202 — APPROVED** [review]

**Pull request merged**